modules/base/rendering/renderableprism.cpp

```diff
@@ -0,0 +1,354 @@
+        "Color",
+        "Color",
+        "This value determines the RGB color for the line."
+    };
+
+    constexpr openspace::properties::Property::PropertyInfo LengthInfo = {
+        "Length",
+        "Length",
+        "The length of the prism in meters."
+    };
+
+    // Generate vertices around the unit circle on the XY-plane
+    void getUnitCircleVertices(std::vector<float>& vertices, int sectorCount) {
+        float sectorStep = 2 * M_PI / sectorCount;
+        float sectorAngle;  // in radians
+
+        for (int i = 0; i < sectorCount; ++i) {
+            sectorAngle = i * sectorStep;
+            vertices.push_back(cos(sectorAngle)); // x
+            vertices.push_back(sin(sectorAngle)); // y
+        }
+    }
+
+    struct [[codegen::Dictionary(RenderablePrism)]] Parameters {
+        // [[codegen::verbatim(SegmentsInfo.description)]]
+        int segments;
+
+        // [[codegen::verbatim(LinesInfo.description)]]
+        std::optional<int> lines;
+
+        // [[codegen::verbatim(RadiusInfo.description)]]
+        std::optional<float> radius;
+
+        // [[codegen::verbatim(LineWidthInfo.description)]]
+        std::optional<float> lineWidth;
+
+        // [[codegen::verbatim(LineColorInfo.description)]]
+        std::optional<glm::vec3> color [[codegen::color()]];
+
+        // [[codegen::verbatim(LengthInfo.description)]]
+        std::optional<float> length;
+    };
+#include "renderableprism_codegen.cpp"
+} // namespace
+
+namespace openspace {
+
+documentation::Documentation RenderablePrism::Documentation() {
+    documentation::Documentation doc = codegen::doc<Parameters>("base_renderable_prism");
+    return doc;
+}
+
+RenderablePrism::RenderablePrism(const ghoul::Dictionary& dictionary)
+    : Renderable(dictionary)
+    , _nShapeSegments(SegmentsInfo, 6, 3, 32)
+    , _nLines(LinesInfo, 6, 0, 32)
+    , _radius(RadiusInfo, 10.f, 0.f, 3.0e12f)
+    , _lineWidth(LineWidthInfo, 1.f, 1.f, 20.f)
+    , _lineColor(LineColorInfo, glm::vec3(1.f), glm::vec3(0.f), glm::vec3(1.f))
+    , _length(LengthInfo, 20.f, 1.f, 3.0e12f)
+{
+    const Parameters p = codegen::bake<Parameters>(dictionary);
+
+    _radius.setViewOption(properties::Property::ViewOptions::Logarithmic);
+    _lineColor.setViewOption(properties::Property::ViewOptions::Color);
+    _length.setViewOption(properties::Property::ViewOptions::Logarithmic);
+
+    _nShapeSegments.onChange([&]() { _prismIsDirty = true; });
+    _nLines.onChange([&]() { _prismIsDirty = true; });
+    _radius.onChange([&]() { _prismIsDirty = true; });
+    _length.onChange([&]() { _prismIsDirty = true; });
+
+    _nShapeSegments = p.segments;
+    _nLines = p.lines.value_or(_nShapeSegments);
+    _radius = p.radius.value_or(_radius);
+    _lineWidth = p.lineWidth.value_or(_lineWidth);
+    _lineColor = p.color.value_or(_lineColor);
+    _length = p.length.value_or(_length);
+
+    addProperty(_nShapeSegments);
```

Comment:
Personally, I prefer to group the functions by property, rather than by function:
So
```cpp
_radius.onChange([&]() { _prismIsDirty = true; });
_radius = p.radius.value_or(_radius);
addProperty(_radius);

_lineColor.setViewOption(properties::Property::ViewOptions::Color);
addProperty(_lineColor);
```
etc.

But personal preference

Reply:
I have the same preference